README.md
ADSPLACE Core integration/staging repository
=====================================

[![Build Status](https://travis-ci.org/ADSPLACE-Project/ADSPLACE.svg?branch=master)](https://travis-ci.org/ADSPLACE-Project/ADSPLACE) [![GitHub version](https://badge.fury.io/gh/ADSPLACE-Project%2FADSPLACE.svg)](https://badge.fury.io/gh/ADSPLACE-Project%2FADSPLACE)

ADSPLACE is an open source crypto-currency focused on fast private transactions with low transaction fees & environmental footprint.  It utilizes a custom Proof of Stake protocol for securing its network and uses an innovative variable seesaw reward mechanism that dynamically balances 90% of its block reward size between masternodes and staking nodes and 10% dedicated for budget proposals. The goal of ADSPLACE is to achieve a decentralized sustainable crypto currency with near instant full-time private transactions, fair governance and community intelligence.
- Anonymized transactions using the [_Zerocoin Protocol_](https://adsplace.com/zADS).
- Fast transactions featuring guaranteed zero confirmation transactions, we call it _SwiftX_.
- Decentralized blockchain voting utilizing Masternode technology to form a DAO. The blockchain will distribute monthly treasury funds based on successful proposals submitted by the community and voted on by the DAO.

More information at [synergy-work.com](https://adsplace.com) Visit our ANN thread at [BitcoinTalk](http://www.bitcointalk.org/index.php?topic=1262920)

### Coin Specs
<table>
<tr><td>Algo</td><td>Quark</td></tr>
<tr><td>Block Time</td><td>60 Seconds</td></tr>
<tr><td>Difficulty Retargeting</td><td>Every Block</td></tr>
<tr><td>Max Coin Supply (PoW Phase)</td><td>500,300 ADS</td></tr>
<tr><td>Max Coin Supply (PoS Phase)</td><td>Infinite</td></tr>
<tr><td>Premine</td><td>500,000 ADS*</td></tr>
</table>

*500,000 ADS Premine was burned in block 1

### Reward Distribution

<table>
<th colspan=4>Genesis Block</th>
<tr><th>Block Height</th><th>Reward Amount</th><th>Notes</th></tr>
<tr><td>1</td><td>500,000 ADS</td><td>Initial Pre-mine, burnt in block 1</a></td></tr>
</table>

### PoW Rewards Breakdown

<table>
<th>Block Height</th><th>Masternodes</th><th>Miner</th><th>Budget</th>
<tr><td>1</td><td>0% (0 ADS)</td><td>100% (500000 ADS)</td><td>N/A</td></tr>
<tr><td>1-500</td><td>0% (0 ADS)</td><td>100% (1 ADS)</td><td>10% (0.1 ADS)</td></tr>
</table>

### PoS Rewards Breakdown

<table>
<thead>
                    <tr>
                        <th scope="col">PHASE FRAME</th>
                        <th scope="col">BLOCK</th>
                        <th scope="col">REWAR PER BLOCK</th>
                        <th scope="col">MASTERNODE REWARD</th>
                        <th scope="col">STAKlNG REWARD</th>
                        <th scope="col">ROUND TIME</th>
                    </tr>
                </thead>
				<tbody>
<tr>
                        <td>1</td>
                        <td>501-10,000</td>
                        <td>1,O</td>
                        <td>0,8</td>
                        <td>0,2</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>2</td>
                        <td>10,001-19,999</td>
                        <td>1,5</td>
                        <td>1,2</td>
                        <td>0,3</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>3</td>
                        <td>20,000-29,999</td>
                        <td>2,0</td>
                        <td>1,6</td>
                        <td>0,4</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>4</td>
                        <td>30,000-39,999</td>
                        <td>2,5</td>
                        <td>2</td>
                        <td>0,5</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>5</td>
                        <td>40,000-49,999</td>
                        <td>3,0</td>
                        <td>2,4</td>
                        <td>0,6</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>6</td>
                        <td>50,000-59,999</td>
                        <td>3,5</td>
                        <td>2,8</td>
                        <td>0,7</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>7</td>
                        <td>60,000-69,999</td>
                        <td>4,0</td>
                        <td>3,2</td>
                        <td>0,8</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>8</td>
                        <td>70,000-79,999</td>
                        <td>4,5</td>
                        <td>3,6</td>
                        <td>0,9</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>9</td>
                        <td>80,000-89,999</td>
                        <td>5,0</td>
                        <td>4</td>
                        <td>1</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>10</td>
                        <td>90,000-99,999</td>
                        <td>5,5</td>
                        <td>4,4</td>
                        <td>1,1</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>11</td>
                        <td>100,000-109,999</td>
                        <td>6,0</td>
                        <td>4,8</td>
                        <td>1,2</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>12</td>
                        <td>110,000-119,999</td>
                        <td>6,5</td>
                        <td>5,2</td>
                        <td>1,3</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>13</td>
                        <td>120,000-129,999</td>
                        <td>7,0</td>
                        <td>5,6</td>
                        <td>1,4</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>14</td>
                        <td>130,000-139,999</td>
                        <td>7,5</td>
                        <td>6</td>
                        <td>1,5</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>15</td>
                        <td>140,000-149,999</td>
                        <td>8,0</td>
                        <td>6,4</td>
                        <td>1,6</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>16</td>
                        <td>150,000-159,999</td>
                        <td>8,5</td>
                        <td>6,8</td>
                        <td>1,7</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>17</td>
                        <td>160,000-169,999</td>
                        <td>9,0</td>
                        <td>7,2</td>
                        <td>1,8</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>18</td>
                        <td>170,000-179,999</td>
                        <td>8,9</td>
                        <td>7,2</td>
                        <td>1,8</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>19</td>
                        <td>180,000-189,999</td>
                        <td>8,8</td>
                        <td>7,04</td>
                        <td>1,76</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>20</td>
                        <td>190,000-199,999</td>
                        <td>8,7</td>
                        <td>6,96</td>
                        <td>1,74</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>21</td>
                        <td>200,000-209,999</td>
                        <td>8,6</td>
                        <td>6,88</td>
                        <td>1,72</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>22</td>
                        <td>210,000-219,999</td>
                        <td>8,5</td>
                        <td>6,8</td>
                        <td>1,7</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>23</td>
                        <td>220,000-229,999</td>
                        <td>8,4</td>
                        <td>6,72</td>
                        <td>1,68</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>24</td>
                        <td>230,000-239,999</td>
                        <td>8,3</td>
                        <td>6,64</td>
                        <td>1,66</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>25</td>
                        <td>240,000-249,999</td>
                        <td>8,2</td>
                        <td>6,56</td>
                        <td>1,64</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>26</td>
                        <td>270,000-279,999</td>
                        <td>8,1</td>
                        <td>6,48</td>
                        <td>1,62</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>27</td>
                        <td>260,000-269,999</td>
                        <td>8</td>
                        <td>6,4</td>
                        <td>1,6</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>28</td>
                        <td>270,000-279,999</td>
                        <td>7,9</td>
                        <td>6,32</td>
                        <td>1,58</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>29</td>
                        <td>280,000-289,999</td>
                        <td>7,8</td>
                        <td>6,24</td>
                        <td>1,56</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>30</td>
                        <td>290,000-299,999</td>
                        <td>7,7</td>
                        <td>6,16</td>
                        <td>1,54</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>31</td>
                        <td>300,000-309,999</td>
                        <td>7,6</td>
                        <td>6,08</td>
                        <td>1,52</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>32</td>
                        <td>310,000-319,999</td>
                        <td>7,5</td>
                        <td>6</td>
                        <td>1,5</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>33</td>
                        <td>320,000-329,999</td>
                        <td>7,4</td>
                        <td>5,92</td>
                        <td>1,48</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>34</td>
                        <td>330,000-339,999</td>
                        <td>7,3</td>
                        <td>5,84</td>
                        <td>1,46</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>35</td>
                        <td>340,000-349,999</td>
                        <td>7,2</td>
                        <td>5,76</td>
                        <td>1,44</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>36</td>
                        <td>350,000-359,999</td>
                        <td>7,1</td>
                        <td>5,76</td>
                        <td>1,42</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>37</td>
                        <td>360,000-369,999</td>
                        <td>7</td>
                        <td>5,6</td>
                        <td>1,4</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>38</td>
                        <td>370,000-379,999</td>
                        <td>6,9</td>
                        <td>5,52</td>
                        <td>1,38</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>39</td>
                        <td>380,000-389,999</td>
                        <td>6,8</td>
                        <td>5,44</td>
                        <td>1,36</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>40</td>
                        <td>390,000-399,999</td>
                        <td>6,7</td>
                        <td>5,36</td>
                        <td>1,34</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>41</td>
                        <td>400,000-409,999</td>
                        <td>6,6</td>
                        <td>5,28</td>
                        <td>1,32</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>42</td>
                        <td>410,000-419,999</td>
                        <td>6,5</td>
                        <td>5,2</td>
                        <td>1,3</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>43</td>
                        <td>420,000-429,999</td>
                        <td>6,4</td>
                        <td>5,12</td>
                        <td>1,28</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>44</td>
                        <td>430,000-439,999</td>
                        <td>6,3</td>
                        <td>5,04</td>
                        <td>1,26</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>45</td>
                        <td>440,000-449,999</td>
                        <td>6,2</td>
                        <td>4,96</td>
                        <td>1,24</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>46</td>
                        <td>450,000-459,999</td>
                        <td>6,1</td>
                        <td>4,88</td>
                        <td>1,22</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>47</td>
                        <td>460,000-469,999</td>
                        <td>6</td>
                        <td>4,8</td>
                        <td>1,2</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>48</td>
                        <td>470,000-479,999</td>
                        <td>5,9</td>
                        <td>4,72</td>
                        <td>1,18</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>49</td>
                        <td>480,000-489,999</td>
                        <td>5,8</td>
                        <td>4,64</td>
                        <td>1,16</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>50</td>
                        <td>490,000-499,999</td>
                        <td>5,7</td>
                        <td>4,56</td>
                        <td>1,14</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>51</td>
                        <td>500,000-509,999</td>
                        <td>5,6</td>
                        <td>4,48</td>
                        <td>1,12</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>52</td>
                        <td>510,000-519,999</td>
                        <td>5,5</td>
                        <td>4,4</td>
                        <td>1,1</td>
                        <td>1WEEK</td>
                    </tr>
                    <tr>
                        <td>53</td>
                        <td>520,000-onwards</td>
                        <td>5</td>
                        <td>4</td>
                        <td>1</td>
                        <td>1WEEK</td>
                    </tr>
					</tbody>
</table>

# How to setup ADSPLACE masternode on Ubuntu
Shell script to install or update [ADSPLACE Masternode](https://adsplace.io) on a Linux server running Ubuntu 18.04.
***

## VPS installation
```
wget -q https://github.com/michailduzhanski/ADSPLACE/releases/download/1.0/ads-mn-install.sh
bash ads-mn-install.sh
```
***

## Desktop wallet setup

After the Masternode is up and running, you need to configure the desktop wallet accordingly. Here are the steps:
1. Open the ADSPLACE Desktop Wallet.
2. Go to RECEIVE and create a New Address: **MN1**
3. Send **1000** ADS to **MN1**.
4. Wait for 15 confirmations.
5. Go to **Help -> "Debug Window - Console"**
6. Type the following command: **masternode outputs**
7. Go to  **Tools -> "Open Masternode Configuration File"**
8. Add the following entry:
```
Alias Address Privkey TxHash TxIndex
```
* Alias: **MN1**
* Address: **VPS_IP:PORT**
* Privkey: **Masternode Private Key**
* TxHash: **First value from Step 6**
* TxIndex:  **Second value from Step 6**
9. Save and close the file.
10. Go to **Masternode Tab**. If you tab is not shown, please enable it from: **Settings - Options - Wallet - Show Masternodes Tab**
11. Click **Update status** to see your node. If it is not shown, close the wallet and start it again. Make sure the wallet is unlocked.
12. Select your MN and click **Start Alias** to start it.
13. Alternatively, open **Debug Console** and type:
```
startmasternode alias false MN1
```
14. Login to your VPS and check your masternode status by running the following command to confirm your MN is running:
```
adsplace-cli masternode status
```
***

## Usage:
```
adsplace-cli masternode status #To check your MN status
adsplace-cli getinfo #To get general info such as ADSPLACE version and current block numnber
adsplace-cli mnsync status #To check if your MN is synced.
```
Also, if you want to check/start/stop **ADSPLACE**, run one of the following commands as **root**:

```
systemctl status ADSPLACE #To check if ADSPLACE service is running
systemctl start ADSPLACE #To start ADSPLACE service
systemctl stop ADSPLACE #To stop ADSPLACE service
systemctl is-enabled ADSPLACE #To check if ADSPLACE service is enabled on boot
```
***


doc/release-notes.md
(note: this is a temporary file, to be added-to by anybody, and moved to release-notes at release time)

ADSPLACE Core version *version* is now available from:  <https://github.com/adsplace-project/adsplace/releases>

This is a new major version release, including various bug fixes and performance improvements, as well as updated translations.

Please report bugs using the issue tracker at github: <https://github.com/adsplace-project/adsplace/issues>


Mandatory Update
==============


How to Upgrade
==============

If you are running an older version, shut it down. Wait until it has completely shut down (which might take a few minutes for older versions), then run the installer (on Windows) or just copy over /Applications/ADSPLACE-Qt (on Mac) or adsplaced/adsplace-qt (on Linux).


Compatibility
==============

ADSPLACE Core is extensively tested on multiple operating systems using the Linux kernel, macOS 10.8+, and Windows 7 and later.

Microsoft ended support for Windows XP on [April 8th, 2014](https://www.microsoft.com/en-us/WindowsForBusiness/end-of-xp-support), No attempt is made to prevent installing or running the software on Windows XP, you can still do so at your own risk but be aware that there are known instabilities and issues. Please do not report issues about Windows XP to the issue tracker.

ADSPLACE Core should also work on most other Unix-like systems but is not frequently tested on them.

### :exclamation::exclamation::exclamation: MacOS 10.13 High Sierra :exclamation::exclamation::exclamation:

**Currently there are issues with the 3.x gitian release on MacOS version 10.13 (High Sierra), no reports of issues on older versions of MacOS.**

 
Notable Changes
==============


*version* Change log
==============

Detailed release notes follow. This overview includes changes that affect behavior, not code moves, refactors and string updates. For convenience in locating the code changes and accompanying discussion, both the pull request and git merge commit are mentioned.

### Core Features

### Build System
 
### P2P Protocol and Network Code

### GUI
 
### RPC/REST

### Wallet
 
### Miscellaneous
 
 
## Credits

Thanks to everyone who directly contributed to this release:


As well as everyone that helped translating on [Transifex](https://www.transifex.com/projects/p/adsplace-project-translations/), the QA team during Testing and the Node hosts supporting our Testnet.


src/bip38.h
// Copyright (c) 2017 The ADSPLACE developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#ifndef BITCOIN_BIP38_H
#define BITCOIN_BIP38_H

#include "pubkey.h"
#include "uint256.h"

#include <string>


/** 39 bytes - 78 characters
 * 1) Prefix - 2 bytes - 4 chars - strKey[0..3]
 * 2) Flagbyte - 1 byte - 2 chars - strKey[4..5]
 * 3) addresshash - 4 bytes - 8 chars - strKey[6..13]
 * 4) Owner Entropy - 8 bytes - 16 chars - strKey[14..29]
 * 5) Encrypted Part 1 - 8 bytes - 16 chars - strKey[30..45]
 * 6) Encrypted Part 2 - 16 bytes - 32 chars - strKey[46..77]
 */

void DecryptAES(uint256 encryptedIn, uint256 decryptionKey, uint256& output);

void ComputePreFactor(std::string strPassphrase, std::string strSalt, uint256& prefactor);

void ComputePassfactor(std::string ownersalt, uint256 prefactor, uint256& passfactor);

bool ComputePasspoint(uint256 passfactor, CPubKey& passpoint);

void ComputeSeedBPass(CPubKey passpoint, std::string strAddressHash, std::string strOwnerSalt, uint512& seedBPass);

void ComputeFactorB(uint256 seedB, uint256& factorB);

std::string BIP38_Encrypt(std::string strAddress, std::string strPassphrase, uint256 privKey, bool fCompressed);
bool BIP38_Decrypt(std::string strPassphrase, std::string strEncryptedKey, uint256& privKey, bool& fCompressed);

std::string AddressToBip38Hash(std::string address);

#endif // BIP38_H
